storagenode/pieces/readwrite.go

```diff
@@ -5,25 +5,81 @@ package pieces
 
 import (
 	"context"
+	"encoding/binary"
 	"hash"
 	"io"
 
+	"github.com/gogo/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/pkcrypto"
 	"storj.io/storj/storage"
 )
 
+const (
+	// V1PieceHeaderReservedArea is the amount of space to be reserved at the beginning of
+	// pieces stored with storage.FormatV1 or greater. Serialized piece headers should be
+	// written into that space, and the remaining space afterward should be zeroes.
+	// V1PieceHeaderReservedArea includes the size of the framing field
+	// (v1PieceHeaderFrameSize). It has a constant size because:
+	//
+	//  * We do not anticipate needing more than this.
+	//  * We will be able to sum up all space used by a satellite (or all satellites) without
+	//    opening and reading from each piece file (stat() is faster than open()).
```

Comment:
I think we should clarify here that the _reason_ the constant header size allows us to do this is because we can determine the `FormatVersion` via the filename, meaning we then know the guaranteed header size, thus we can calculate the actual file size without doing a file read.

Reply:
This bullet item is referring only to pieces with V1-style headers. What you say is certainly true, and is indeed why the same benefit can be had in the presence of a mix of V0 and V1 piece files, but the salient point in this particular place is only that the fixed header size allows faster piece size determination for V1 pieces.